### PR TITLE
Implementation of the user interface for the tictactoe project - Milestone 2

### PR DESCRIPTION
**Pull request by jadx2**
Created a Player class that will take care of the creation of both players and token assignment. In the main, the players input their names, but only the "player1" can choose a token. "player2" receives automatically the remaining token. 
The game starts displaying the board in the terminal and asking for the "player1" to make a move. This move is going to be validated by a method in the board class to make sure that the input for the move is a number between 1 and 9. If the input is correct then the next player makes a move.
This loop will keep going until there is a winner or a draw.

### CODE REVIEW

**Review by MahmoudBakr23 — CHANGES_REQUESTED** [review]
### **Status: Changes Required ♻**
<hr>
Hello george-shammar and jadx2, well done working on this milestone. However, there are some changes to do before moving on to the next milestone.
<hr>

- [x] Please make a more descriptive pull request header
- [ ] Please make a more descriptive pull request body it could be like: first, it asks for the player 1 name and token, and then the player 2 two ...
- [x] Make sure that the milestone meets all the instructions, the game has to determine the winner, of course, it's a mockery but it has to show at the end of it that the winner is X (for example)
logic

- [x] In order to make a winner, the game has to stop at some point, so you should make a variable called game_on = true and while it's true the game goes on, and once there's a winner the game stops.
Screenshot
- [x] The game should go on above more than just two moves, and along with it, you should check if the number has been selected before or not! if so, it'd inform the player that it's an invalid move
- [x] game displays board after the player move (without computing which spots on the board are taken)
<hr>
Please make sure to apply these changes before submitting a re-review and remember the goal of this milestone is to prepare and set your mind on what you're going to do with the logic in the next milestone
<hr>
Happy coding 💻👨‍💻

**Review by javitocor — CHANGES_REQUESTED** [review]
# STATUS CHANGES REQUIRED 🛑 

Good job team implementing the changes! 👍 
However, we still need something else to get this project approved 💪 
Kindly check my comments below 👇  apply the changes and resubmit your project  for a new code review ⏩ 
Good luck! 😺 
## Mandatory changes
- Kindly fix the linter errors
- Your loop is infinite, it should not, make it finish once the players have reached 9 turns
- Game should ask the player to select from available moves
- Game should inform the player if the selected move is a winning move (without computing the winning move)
- Game should inform the player if the selected move is a draw move (without computing draw)

**Review by ebeagusamuel — CHANGES_REQUESTED** [review]
# Changes Required :recycle: 
### Hello,
##### For this milestone, you are expected to only mock the basic flow of the game. This means implementing a user interface that displays prompts for users (print/puts) and collects user input (gets.chomp). No game logic (for example checking who is a winner) is required in this milestone - instead, students can just display "The winner is X" no matter who won. 
##### For this milestone, you don't need to have any code in the lib folder, the code for mocking the game should be in the main.rb file in the bin folder.
##### There is a bug in your code that is causing it to break when a player tries to make a move.
#### Please check your code and make the changes stated above and once you have done that request another review.
slack(@ebeagusamuel)

File: bin/main [hunk 1]
```diff
@@ -1,2 +1,52 @@
...
+player2 = gets.chomp
+
+players = { player1 => 'X', player2 => 'O' }
+
+puts "#{player1} is #{players[player1]}"
+puts "#{player2} is #{players[player2]}"
+
+# game_on = true
+# winner = false
+# draw = false
```
Comment (enaburekhan) [inline bin/main:18]: It's a good practice not to leave commented  out code in your codebase. If they are not required in this milestone, pls remove them.

File: bin/main [hunk 2]
```diff
@@ -1,2 +1,52 @@
...
+board = <<-HERDOC
+                   1 | 2 | 3
+                  ---+---+---
+                   4 | 5 | 6
+                  ---+---+---
+                   7 | 8 | 9
+HERDOC
+
+9.times do
```
Comment (enaburekhan) [inline bin/main:27]: Kindly change the number "9" in your loop. Using numbers to create a loop are been discourage in this milestone.
Instead, you can use a while loop or any other type of loop that is convenient for you. I can see you uncomment out 
`game_on = true` and `game_on = false`. you can use them to initialize and terminate your game :blush:

**Review by enaburekhan — CHANGES_REQUESTED** [review]
# Request changes
**Hi team,**
**Good job so far. However, there are still some changes to be made in other to get this project approved.**

- There is still a bug in your code as it keeps displaying only one player when playing the game.

- Kindly check this list for guidance:

image

- Remove all commented out code from your codebase. 

Kindly make the above changes and submit for another review :blush:.

You can reach me on slack @ enaburekhaneric.
Happy coding!!!

**Review by LamiaSristy — APPROVED** [review]
# Status: Approved 👍 👍 

Hello Team @jadx2 and @george-shammar. ✋ 

Great work, implementing the changes mentioned by my former TSEs. 🛠️ 

You can merge the branch now, good luck with your next milestone 🚀.

Feel free to reach me at @lamiasristy

**Pull request merged**